components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/deployers/FaultyDeployer1.java

```diff
@@ -79,7 +93,11 @@ public void undeploy(Object key) throws CarbonDeploymentException {
 
     public String update(Artifact artifact) throws CarbonDeploymentException {
         logger.info("Updating : " + artifact.getName());
-        sample1Updated = true;
+        if (artifact.getName().contains("sample1.txt")) {
```

Comment:
Since these are test artifacts if there is a NPE then the test will fail. So do we need to do a null check?